nipoppy/workflow/proc_pipe/tractoflow/run_tractoflow.py

```diff
@@ -636,41 +661,64 @@ def run(participant_id, global_configs, session_id, output_dir, use_bids_filter,
-    
+
     ## compose tractoflow arguments
     TRACTOFLOW_CMD=f""" --input {tractoflow_nxtf_inp} --output_dir {tractoflow_out_dir} --run_t1_denoising --run_gibbs_correction --encoding_direction {phase} --readout {readout} --dti_shells "0 {dti_use}" --fodf_shells "0 {odf_use}" --sh_order {sh_order} --profile {profile} --processes {ncore}"""
 
     ## TractoFlow arguments can be printed multiple ways that appear consistent with the documentation but are parsed incorrectly by nextflow.
     ## .nexflow.log (a run log that documents what is getting parsed by nexflow) shows additional quotes being added around the dti / fodf parameters. Something like: "'0' '1000'"
     ## Obviously, this breaks the calls that nextflow tries to make at somepoint (typically around Normalize_DWI) because half the command becomes an unfinished text block from the mysteriously added quotes.
     ## I don't know if the problem is python printing unhelpful/inaccurate text to the user or if nextflow can't parse its own input arguments correctly.
-    
+
     ## add resume option if working directory is not empty
     if not len(os.listdir(tractoflow_work_dir)) == 0:
         TRACTOFLOW_CMD = TRACTOFLOW_CMD + " -resume"
-    
+
     ## build command line call
-    CMD_ARGS = NEXTFLOW_CMD + TRACTOFLOW_CMD 
-    
+    CMD_ARGS = NEXTFLOW_CMD + TRACTOFLOW_CMD
+
     ## log what is called
     logger.info("-"*75)
     logger.info(f"Running TractoFlow for participant: {participant_id}")
 
-    ## singularity 
+    ## singularity
     SINGULARITY_CMD=f"{SINGULARITY_COMMAND} exec --cleanenv -H {nextflow_logdir} -B {nextflow_logdir}:/nextflow -B {LOGDIR} -B {output_dir} {SINGULARITY_TRACTOFLOW}"
 
     CMD=SINGULARITY_CMD + " " + CMD_ARGS
     logger.info("+"*75)
     logger.info(f"Command passed to system:\n\n{CMD}\n")
     logger.info("+"*75)
-    
+
     ## there's probably a better way to try / catch the .run() call here
     try:
         logger.info('Attempting Run')
         tractoflow_proc = subprocess.run(CMD, shell=True)
+
+        # do clean-up
+
+        # check if tractogram is a valid file
+        trk = nib.streamlines.load(Path(tractoflow_out_dir, 'PFT_Tracking', f"{participant_id}__pft_tracking_prob_wm_seed_0.trk"))
+
+        # if the the last file created (tractogram) can be validly loaded,
+        # processing is done.
+        if trk:
+
+            # find every file in the output path (this should grab symlinks)
+            out_files = Path(tractoflow_out_dir).rglob("*")
+
+            # convert symlinks to real files
+            for out in out_files:
+                move_target_to_symlink(out)
+
+            # remove tractoflow_input_dir
+            shutil.rmtree(tractoflow_input_dir)
```

Comment:
Is this removing the entire `derivatives/tractoflow/<version>/input` directory? Probably not the best thing to do since it might break parallel runs, unless I'm misunderstanding this.

I see that there is also a `tractoflow_subj_dir` -- might be better to remove this one (and its parent) instead.

Reply:
Good catch, that should have been `tractoflow_nxtf_inp`

They have to be nested deeply to prevent conflicts with tractoflows parsing "help".